Sources/SwiftParser/Statements.swift

```diff
@@ -572,7 +572,7 @@ extension Parser {
         type = nil
       }
     } else {
-      (pattern, type) = self.parseTypedPattern()
+      (pattern, type) = self.parseTypedPattern(allowRecover: false)
```

Comment:
Why not always recover?

Reply:
For the case the user forgot the `in` in `for item in list {…}` statements we would suggest the user to insert a `:` instead of an `in` because `list` would be assumed as the type.